crates/pallet-domains/src/staking_epoch.rs

```diff
@@ -380,12 +215,13 @@ pub(crate) fn do_finalize_domain_staking<T: Config>(
 
         Ok(previous_epoch)
     })
-    .map_err(Error::FinalizeDomainPendingTransfers)
+    .map_err(Error::FinalizeDomainEpochStaking)
 }
 
-fn finalize_operator_pending_transfers<T: Config>(
+pub(crate) fn do_finalize_operator_epoch_staking<T: Config>(
+    domain_id: DomainId,
     operator_id: OperatorId,
-    domain_block_number: DomainBlockNumberFor<T>,
+    previous_epoch: EpochIndex,
```

Comment:
I see, this is probably the source of deviation that led to this and other related comments.

>As soon as the current epoch is finished, the total stake will include rewards from the previous epoch. All the deposits from the previous epoch will be returned to the nominator as is and all withdrawals are included with know_shares to calculate the final stake amount.

The way I understand is that epoch finalization is used to handle the deposit/withdraw requests and reward happen in the current epoch and even for the de-registered/slashed operator there is such request/reward that need to handle. The current implementation seems to manually handle these in `do_unlock_operator` and `do_finalize_slashed_operators` instead of reusing the existing infra.

Even we don't handle these operators during epoch transition, it may simplify the `do_unlock_operator`/`do_finalize_slashed_operators` a lot by manually calling `do_finalize_operator_epoch_staking` within `do_unlock_operator`/`do_finalize_slashed_operators`

Reply:
We don't have to epoch transition for these operators, calling `do_finalize_operator_epoch_staking` inside `do_unlock_operator/do_finalize_slashed_operators` should have the same result (similar to call `do_finalize_operator_epoch_staking` inside `do_finalize_switch_operator_domain`).

I always prefer code that is simpler and easier to reason security even it comes with (minor) performance cost, but yes there is no specific issue with the current implementation thus no strong opinion.